CONTRIBUTING.md

```diff
@@ -0,0 +1,114 @@
+# Contribution Guide
+
+We welcome and encourage community contributions to `rack`.
+
+Please familiarize yourself with the Contribution Guidelines and Project Roadmap before contributing.
+
+There are many ways to help `rack` besides contributing code:
+
+ - Fix bugs or file issues
+ - Improve the [documentation](https://github.com/jrperritt/rack/tree/master/docs)
+
+## Contributing Code
+
+Unless you are fixing a known bug, we **strongly** recommend discussing it with the core team via a GitHub issue or on [Freenode IRC](irc://chat.freenode.net/rackspace) before getting started to ensure your work is consistent with `rack`'s roadmap and architecture.
```

Comment:
Are we actually all on freenode watching over this? I'd rather people just file issues no matter what.